CHALLENGE.md

```diff
@@ -91,7 +91,7 @@ Requirements:
 
 ![search-no-result](.github/search-no-result-state.png)
 
-#### 4.F) Optimize the search functionality for performance.
+#### 4.F) Optimize the search functionality to reduce the number of excessive requests.
```

Comment:
The reason why the challenge says "Optimize the search functionality **for performance.**" it's because the intent was for the candidate to decide how to do so.

I don't see this as a problem, but more like a chance for discussion in the interview :)

Reply:
I agree with Stefano. Keeping some requirements a bit vague gives the chance to us to see what the candidates know and wants to implements about that.

Excessive requests, unnecessary rerender, data storing, etc. 

We don't need and we don't want to spell everything out for them

Reply:
@stefanosusini @ejrcavenago I understand, should we then update the reviewer part of the test to specify that we won't mark these if the candidate has not implemented debounce but another performance improvement?

Reply:
We should judge the overall decision of performance that the candidate has done.

If you have some well thought data storing strategy for search but don't have debounce, well, I think we should still score that negatively. Debounce is VERY important, data storage (or other example) not the same.

The fact that the request is open ended, doesn't meant that we don't know what we want to look for. It means that we expect the candidate to know about the potential issues of calling an api every keypress and deal with that, event if we don't tell you.

We're looking always for P3+, not people just out of school.

Reply:
I was more worries with the meaning of "performance".

When I search "optimizing performance in react" in Google I find this (https://reactjs.org/docs/optimizing-performance.html). For the react docs, optimizing performance means to "minimize the number of costly DOM operations required to update the UI". There is not a single mention of debounce or throttling in the "optimizing performance" page.

So the candidate might just think, well, I have improved the performance, and I'm not sure if debounce could count as a performance improvement and the challenge doesn't mention it, so I will just follow the spec. 

Anyway, since you have both different opinions I will close this.